src/wp-admin/includes/update-core.php

```diff
@@ -1113,15 +1113,27 @@ function update_core( $from, $to ) {
 		);
 	}
 
-	// Add a warning when the JSON PHP extension is missing.
-	if ( ! extension_loaded( 'json' ) ) {
+	$required_extensions = array(
+		'json',
+		'mbstring',
+	);
+	$missing_extensions  = array();
+
+	foreach ( $required_extensions as $extension ) {
+		if ( ! extension_loaded( $extension ) ) {
+			$missing_extensions[] = $extension;
+		}
+	}
+
+	// Add a warning when the required PHP extensions are missing.
+	foreach ( $missing_extensions as $missing_extension ) {
```

Comment:
On servers missing multiple required extensions, this will only return the first missing extension. I think you'll need to create the object and use `WP_Error::add` in the loop and return it if it's populated.